src/mpcCoreKit.ts

```diff
@@ -1003,6 +1026,10 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
         }
       }
       await this.handleNewUser(importTssKey);
+      if (importTssKey && isSfaKey) {
+        await this.tkey.addLocalMetadataTransitions({ input: [{ message: ONE_KEY_DELETE_NONCE }], privKey: [new BN(this.state.postBoxKey, "hex")] });
+        if (!this.tkey?.manualSync) await this.tkey?.syncLocalMetadataTransitions();
```

Comment:
The metadata sync is not atomic in this case.
we could wrap the handleNewUser and addLocalMetadatatransition in a atomicSync function to make it atomic

```
this.atomicSync( async () => {
    await this.handleNewUser(importTssKey);
    if (importTssKey && isSfaKey) {
       await this.tkey.addLocalMetadataTransitions({ input: [{ message: ONE_KEY_DELETE_NONCE }], privKey: [new BN(this.state.postBoxKey, "hex")] });
     }
})

Reply:
Done, i have rather moved it inside handleNewUser itself